llarp/util/time.cpp

```diff
@@ -6,15 +6,33 @@ namespace llarp
 {
   using Clock_t = std::chrono::system_clock;
 
-  template < typename Res >
+  template < typename Res, typename Clock >
   static llarp_time_t
   time_since_epoch()
   {
-    return std::chrono::duration_cast< Res >(
-               llarp::Clock_t::now().time_since_epoch())
+    return std::chrono::duration_cast< Res >(Clock::now().time_since_epoch())
         .count();
   }
 
+  static llarp_time_t
+  time_started_at()
+  {
+    const static llarp_time_t started =
+        time_since_epoch< std::chrono::milliseconds, Clock_t >();
```

Comment:
Should use static initialization outside the function (static locals only get initialized when they are first encountered).